static/app/components/charts/onDemandMetricRequest.tsx

```diff
@@ -3,57 +3,16 @@ import {addErrorMessage} from 'sentry/actionCreators/indicator';
 import EventsRequest from 'sentry/components/charts/eventsRequest';
 import {t} from 'sentry/locale';
 import {EventsStats, MultiSeriesEventsStats} from 'sentry/types';
-import {DiscoverDatasets} from 'sentry/utils/discover/types';
-
-function numOfEvents(timeseriesData) {
-  return timeseriesData.data.reduce((acc, item) => {
-    const count = item[1][0].count;
-    return acc + count;
-  }, 0);
-}
-
-function applySampleRate(timeseriesData, sampleRate = 1) {
-  const scaledData = timeseriesData.data.map(([timestamp, value]) => {
-    return [timestamp, value.map(item => ({count: Math.round(item.count / sampleRate)}))];
-  });
-
-  return {
-    ...timeseriesData,
-    isExtrapolatedData: true,
```

Comment:
We need to append this flag (since it is used to control the UI) to the returned `timeseriesData`. We can do that either here or when we mutate the state in `fetchData`.